src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombiner.java

```diff
@@ -68,42 +71,46 @@ private float combineShardScores(
         );
 
         // - sort documents by scores and take first "max number" of docs
-        // create a priority queue of doc ids that are sorted by their combined scores
-        PriorityQueue<Integer> scoreQueue = getPriorityQueueOfDocIds(combinedNormalizedScoresByDocId);
-        // store max score to resulting list, call it now as priority queue will change after combining scores
-        float maxScore = combinedNormalizedScoresByDocId.get(scoreQueue.peek());
+        // create a collection of doc ids that are sorted by their combined scores
+        List<Integer> sortedDocsIds = getPriorityQueueOfDocIds(combinedNormalizedScoresByDocId);
 
         // - update query search results with normalized scores
-        updateQueryTopDocsWithCombinedScores(compoundQueryTopDocs, topDocsPerSubQuery, combinedNormalizedScoresByDocId, scoreQueue);
-        return maxScore;
+        updateQueryTopDocsWithCombinedScores(compoundQueryTopDocs, topDocsPerSubQuery, combinedNormalizedScoresByDocId, sortedDocsIds);
+
+        // return max score
+        if (sortedDocsIds.isEmpty()) {
+            return ZERO_SCORE;
+        }
+        return combinedNormalizedScoresByDocId.get(sortedDocsIds.get(0));
     }
 
-    private PriorityQueue<Integer> getPriorityQueueOfDocIds(final Map<Integer, Float> combinedNormalizedScoresByDocId) {
-        PriorityQueue<Integer> pq = new PriorityQueue<>(
-            (a, b) -> Float.compare(combinedNormalizedScoresByDocId.get(b), combinedNormalizedScoresByDocId.get(a))
-        );
+    @VisibleForTesting
+    protected List<Integer> getPriorityQueueOfDocIds(final Map<Integer, Float> combinedNormalizedScoresByDocId) {
```

Comment:
Method name needs to be updated as well. It is not PriorityQueue anymore.

Reply:
ack